application-analytics-default/src/test/java/com/xwiki/analytics/internal/MostViewedJsonNormaliserTest.java

```diff
@@ -0,0 +1,108 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.analytics.internal;
+
+import java.net.URL;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+import org.xwiki.resource.ResourceReferenceResolver;
+import org.xwiki.resource.ResourceType;
+import org.xwiki.resource.ResourceTypeResolver;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+import org.xwiki.url.ExtendedURL;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+@ComponentTest
+public class MostViewedJsonNormaliserTest
+{
+    @InjectMockComponents
+    private MostViewedJsonNormaliser mostViewedJsonNormaliser;
+
+    @MockComponent
+    private ResourceReferenceResolver<ExtendedURL> resourceReferenceResolver;
+
+    @MockComponent
+    private ResourceTypeResolver<ExtendedURL> resourceTypeResolver;
+
+    @Test
+    public void testObjectJSONResponse() throws Exception
+    {
+        setupMocks("http://localhost:8080/xwiki/bin/view/Analytics/",
+            "http://localhost:8080/xwiki/bin/view/Analytics/Code/Macros/MostViewedPage");
+        assertEquals(responseObjectJSON(), mostViewedJsonNormaliser.normaliseData(getObjectJSON()).toString());
+    }
+
+    @Test
+    public void testArrayJSONResponse() throws Exception
+    {
+        setupMocks("http://localhost:8080/xwiki/bin/view/Analytics/Code/MostViwedPages",
+            "http://localhost:8081/xwiki/bin/view/Main/");
+        assertEquals(responseArrayJSON(), mostViewedJsonNormaliser.normaliseData(getArrayJSONS()).toString());
+    }
+
+    private void setupMocks(String url1, String url2) throws Exception
+    {
+        ExtendedURL extendedURL1 = new ExtendedURL(new URL(url1), null);
+        ExtendedURL extendedURL2 = new ExtendedURL(new URL(url2), null);
+
+        ResourceType resourceType1 = resourceTypeResolver.resolve(extendedURL1, Collections.emptyMap());
+        ResourceType resourceType2 = resourceTypeResolver.resolve(extendedURL2, Collections.emptyMap());
+
+        when(resourceReferenceResolver.resolve(extendedURL1, resourceType1, Collections.emptyMap())).thenReturn(null);
+        when(resourceReferenceResolver.resolve(extendedURL2, resourceType2, Collections.emptyMap())).thenReturn(null);
+    }
+
+    private String getObjectJSON()
+    {
+        return "{\"2023-04\":[],\"2023-05\":[],\"2023-06\":[],\"2023-07\":[{\"label\":\"\\/xwiki\\/bin\\/view\\/Analy"
```

Comment:
I think it would be cleaner and easier to follow to use some JSON files (resources) and load them directly from there.
Found this as example https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/test/java/org/xwiki/web/UorgsuggestPageTest.java#L167 , maybe it helps